asynciterator.js

```diff
@@ -317,7 +392,7 @@ function emitData() {
   while (this._hasListeners('data') && (item = this.read()) !== null)
     this.emit('data', item);
   // Stop draining the source if there are no more `data` listeners
-  if (!this._hasListeners('data')) {
+  if (!this._hasListeners('data') && !this.done) {
```

Comment:
What condition do we catch with this? Was this a bug that already existed, or a new condition introduced because of destroyed?

Reply:
Without this check, the _"should not be listening for the `newListener` event"_ tests fail after calling `destroy`. This is because forcefully destroying the stream can still have generating items in progress, and this waiting-loop will therefore be still running. The way this waiting-loop is implemented at the moment, I don't think there's another way to handle this.